rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,146 @@
+# Stop processing transactions after a certain slot
+
+This PR describes the feature to stop processing transactions after a certain
+slot, to be used in the Berkeley hard fork.
+
+## Summary
+
+Transactions come from a client or the gossip network and are processed by BPs
+and SNARK workers to be included in blocks.
+
+In this RFC, the procedure to stop processing any new transactions after a
+certain slot is described. This is, any blocks produced after that slot will
+include no transaction at all.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. We want this to happen by defining a
+point in time (slot) where the network stops accepting transactions a keeps
+producing blocks free of transactions until it reaches a consensus state.
+
+This feature enables part of this procedure, by allowing users to define a slot
+at which the node will stop accepting any new transactions or including them in
+blocks (in case it is a BP node).
+
+## Detailed design
+
+The procedure to stop processing transactions after a certain slot will act on
+different components of the node, namely, the CLI, the transaction pool, the
+snark pool, and the block producer.
+and the CLI. The procedure will be as follows:
+
+* The CLI will allow a slot at which the node will stop processing transactions
+  to be set. This will be done by setting a new optional flag `--slot-tx-end`
```

Comment:
Is a CLI flag the right place to define this value? I would have thought that the 'runtime configuration' would be a more natural home for this, albeit could be overridden by the flag.

Reply:
Agree with @mrmr1993. If this is a CLI flag, it not only requires every node operator to specify it directly, but any node operators that specify it incorrectly will potentially generate different forks of the network since they disagree on the rules for validating incoming blocks.

Reply:
I agree that this should be introduced in compile time to avoid node operators setting a wrong slot and allowing it to be overridden via runtime configuration/flag.